### ISSUE TITLE
Changing Language Model in QuatzNet5*15

### ISSUE THREAD

**Issue by rose768** [NVIDIA/NeMo#2313]
Hi
I use QuartzNet15-5 for fine-tuning. Based on this [paper](https://arxiv.org/pdf/1910.10261.pdf) I understand that QuartzNet15-5 uses 6-gram and T-XL language models. How can I change the language model in QuartNet15-5? For instance, using Bert.

**Environment overview**

Environment location: Google Colab

Method of NeMo install: !pip install nemo_toolkit[asr]

NeMo version: 1.0.0

Learning Rate: 1e-3

**Environment details**

OS version : "Ubuntu20.04.3 LTS"
PyTorch version : "1.7.1"

**Comment by VahidooX** [comment]
BERT is a masked LM and different from regular Transformer LM. Using a maksed LM model to score a sentence is not as efficient as Transformer. You need to call it X times and X is equal to the number of tokens in a candidate while Transformer just need one pass. For instance, if average of your candidates have 30 tokens, BERT is 30 times slower than Transformer. It makes BERT very expensive for rescoring while the improvement may not be that much significant. There are some papers which have tried to address this issue like this one:

https://assets.amazon.science/cc/54/980cb7d74f93849b49b9d5c42466/masked-language-model-scoring.pdf
